rasa/nlu/training_data/training_data.py

```diff
@@ -111,13 +111,20 @@ def __hash__(self) -> int:
     def sanitize_examples(examples: List[Message]) -> List[Message]:
         """Makes sure the training data is clean.
 
-        removes trailing whitespaces from intent annotations."""
+        removes trailing whitespaces from intent annotations and drops duplicate examples."""
 
         for ex in examples:
             if ex.get("intent"):
                 ex.set("intent", ex.get("intent").strip())
+
             if ex.get("response"):
                 ex.set("response", ex.get("response").strip())
+
+        # this check is necessary because test_markdown_entity_regex() definition assumes order (but test case has no
+        # duplicates)
+        if len(set(examples)) != len(examples):
```

Comment:
You can replace this with `examples = list(set(examples))` without having to perform a separate check.

Reply:
As per the comments, not performing the check will cause `test_markdown_entity_regex()` to fail.

Should we rewrite that test sot that it passes?

Reply:
@msamogh have you had a chance to review this?

Reply:
Sorry for the delay. 

> As per the comments, not performing the check will cause test_markdown_entity_regex() to fail.
That makes sense!

Reply:
Looks good to go!

Reply:
I'd rather avoid the if and change the test

Reply:
@tmbo I've rewritten the test. Not sure if it is the best way though. I'm open to suggestions.